src/memory_pool_tracking.c

```diff
@@ -66,7 +76,7 @@ void umfPoolDestroy(umf_memory_pool_handle_t hPool) {
     }
     // Destroy tracking provider.
     umfMemoryProviderDestroy(hPool->provider);
-    free(hPool);
+    umf_ba_linear_destroy(hPool->pool_linear); // it frees hPool
```

Comment:
You mean: it frees pool_linear?

Reply:
No, it destroys pool_linear and **frees hPool** which was allocated inside pool_linear.

Reply:
Not obvious, but you're right, sorry.